src/apps/bridge/reportBuilder.cpp

```diff
@@ -156,7 +169,7 @@ void ReportBuilderLinkedList::addSampleToElement(report_builder_element_t *eleme
           // We use the element->sample_counter to track within each element how many samples
           // the element has received.
           for (; element->sample_counter < sample_counter; element->sample_counter++) {
-            memcpy(&(static_cast<aanderaa_aggregations_t *>(element->sensor_data))[element->sample_counter++], &NAN_AGG, sizeof(aanderaa_aggregations_t));
+            memcpy(&(static_cast<aanderaa_aggregations_t *>(element->sensor_data))[element->sample_counter], &NAN_AGG, sizeof(aanderaa_aggregations_t));
```

Comment:
this is the bug fix! 

the behavior with the erroneous increment is that we would have an array of 2 samples and accidentally append the second sample (when backfilling) beyond the end of the array like `[NAN sample, NULL(skipped)] NAN (beyond end of the allocated array)`